File: PYTHON/read_ensemble.m
function XENS=read_ensemble(path_data,prefix,NLons,NLats,NVars,EnsSize,Time)


XENS=NaN(NLats,NLons,NVars,EnsSize);  %Analysis ensemble mean

  tstr=num2str(10000+Time);
  tstr=tstr(2:end);
  
  if( strcmp(prefix,'XA') )
      folder='ANALYSIS';
  end
  if( strcmp(prefix,'XB') )
      folder='GUES';
  end
  
  
  for iens=1:EnsSize
    ensstr=num2str(1000+iens);
    ensstr=ensstr(2:end);
    
    file=[path_data '/' folder '/ENS/' prefix '_T' tstr '_M' ensstr];
    nfile=fopen(file,'r','l');
    for ivar=1:NVars
    XENS(:,:,ivar,iens)=flipdim(fread(nfile,[NLons NLats],'single')',1);  
    end
    fclose(nfile);
  end

File: PYTHON/read_ensemble.py
import numpy as np
# Read an ensemble of forecast for a particular time.

def read_ensemble(path_data,prefix,NLons,NLats,NVars,EnsSize,Time):

    # Initialize the ensemble array XBENS(lat,lon,variable,ensemble member)
    XENS = np.full((NLats, NLons, NVars, EnsSize), np.nan)   #Analysis ensemble mean
    tstr = str(10000 + Time) 
    tstr = tstr[1::]
  
    if prefix == 'XA':
        folder = 'ANALYSIS' 
    elif prefix == 'XB':
        folder = 'GUES' 
  
    for iens in range(EnsSize):
        ensstr = str(1000+iens+1) 
        ensstr = ensstr[1::]

        file = path_data + folder + '/ENS/' + prefix + '_T' + tstr + '_M' + ensstr 
        
        for ivar in range(NVars):
            nfile = open(file,'rb') 
            f = np.fromfile(nfile, 'float32')[NLons*NLats*ivar:(1 + ivar)*NLons*NLats].reshape([NLats, NLons])
            XENS[:,:,ivar,iens] = np.flip(f, 0)
    return XENS

  